front/components/vaults/VaultCreateWebsiteModal.tsx

```diff
@@ -1,32 +1,602 @@
+                  />
+                  <Input
+                    placeholder={WEBCRAWLER_MAX_PAGES.toString()}
+                    value={maxPages?.toString() || ""}
+                    onChange={(value) => {
+                      const parsed = parseInt(value);
+                      if (!isNaN(parsed)) {
+                        setMaxPages(parseInt(value));
+                      } else if (value == "") {
+                        setMaxPages(null);
+                      }
+                    }}
+                    showErrorLabel={
+                      maxPages &&
+                      maxPages > WEBCRAWLER_MAX_PAGES &&
+                      maxPages &&
+                      maxPages < 1
+                        ? false
+                        : true
+                    }
+                    error={
+                      (maxPages && maxPages > WEBCRAWLER_MAX_PAGES) ||
+                      (maxPages && maxPages < 1)
+                        ? `Maximum pages must be between 1 and ${WEBCRAWLER_MAX_PAGES}`
+                        : null
+                    }
+                    name="maxPages"
+                  />
+                </Page.Layout>
+              </div>
+              <Page.Layout direction="vertical" gap="md">
+                <Page.H variant="h3">Name</Page.H>
+                <Page.P>Give a name to this Data Source.</Page.P>
+                <Input
+                  placeholder=""
+                  value={dataSourceName}
+                  onChange={(value) => setDataSourceName(value)}
+                  error={dataSourceNameError}
+                  name="dataSourceName"
+                  showErrorLabel={true}
+                  className="text-sm"
+                  disabled={webCrawlerConfiguration !== null}
+                />
+              </Page.Layout>
+
+              <div className="flex">
+                <Button
+                  label="Advanced settings"
+                  variant="secondary"
+                  onClick={() => {
+                    setAdvancedSettingsOpened(true);
+                  }}
+                ></Button>
+              </div>
+              {webCrawlerConfiguration && dataSourceId && (
+                <div className="flex py-16">
+                  <Button
+                    variant="secondaryWarning"
+                    icon={TrashIcon}
+                    label={"Delete this website"}
+                    onClick={() => {
+                      setIsDeleteModalOpen(true);
+                    }}
+                  />
+                  <DeleteDataSourceDialog
+                    handleDelete={handleDelete}
+                    isOpen={isDeleteModalOpen}
+                    setIsOpen={setIsDeleteModalOpen}
+                    dataSourceUsage={0} // TODO: get usage
+                  />
+                </div>
+              )}
+            </Page.Layout>
+          </div>
```

Comment:
Would be nice to split this big component into smaller components within this same file.

Reply:
There is a new design for that, but with Ed we decided to start with this one (the existing one)